bundle/edu.gemini.spModel.core/src/main/scala/edu/gemini/spModel/core/Ephemeris.scala

```diff
@@ -1,9 +1,46 @@
 package edu.gemini.spModel.core
 
-import scalaz._, Scalaz._
+import java.util.TimerTask
+import java.util.logging.Logger
 
-final case class Ephemeris(site: Site, data: Long ==>> Coordinates) {
+import scalaz._, Scalaz._
 
+final class Ephemeris(val site: Site, val compressedData: Deflated[List[(Long, Float, Float)]]) extends Serializable {
+
+  // Internal state for managing the inflated ephemeris data.
+  @transient @volatile private var _data: Long ==>> Coordinates = null
+  @transient @volatile private var _task: TimerTask = null
+
+  /**
+   * A map from time to coordinates. This value is stored in compressed form and is deflated (and
+   * cached for a while) on demand. The deflated value is transient, and is thus not serialized.
+   */
+  def data: Long ==>> Coordinates = synchronized {
+    import Ephemeris.{ timer, logger }
+
+    // Decompress if needed
+    if (_data == null) {
+      logger.info("Inflating compressed ephemeris.")
+      _data = ==>>.fromList(compressedData.inflate.map { case (t, r, d) =>
+        t -> Coordinates.fromDegrees(r, d).getOrElse(sys.error(s"corrupted ephemeris data: $t $r $d"))
+      })
+    }
+
+    // Reset the cache timer to a point the future
+    if (_task != null) _task.cancel()
+    _task = new TimerTask {
+      def run(): Unit = {
+        logger.info("Discarding inflated ephemeris.")
+        Ephemeris.this.synchronized(_data = null)
+      }
+    }
```

Comment:
I think the vast majority of the time we're just asking for the coordinates at the scheduling block start time.  I would like to experiment with this PR as it is vs. one additional tweak.  The coordinate lookup method could just remember the time and coordinate of the last value it was asked to find.  That way if the next request is for the same time it can return the coordinates without inflating the ephemeris data.  If the request is for a different time then it can go ahead and inflate and remember the new (time, coordinate) pair.  My suspicion is that this will save a lot of churn that will otherwise be caused by P2 checking and ODB queries.